src/test/java/org/opensearch/knn/index/OpenSearchIT.java

```diff
@@ -399,4 +399,71 @@ public void testExistsQuery() throws IOException {
 
         assertEquals(3, parseTotalSearchHits(EntityUtils.toString(response.getEntity())));
     }
+
+    public void testIndexingVectorValidation_updateVectorWithNull() throws Exception {
+        Settings settings = Settings.builder()
+                .put(getKNNDefaultIndexSettings())
+                .build();
+
+        createKnnIndex(INDEX_NAME, settings, createKnnIndexMapping(FIELD_NAME, 4));
+
+        // valid case with 4 dimension
+        final Float[] vectorForDocumentOne = {6.0f, 7.0f, 8.0f, 9.0f};
+        final String docOneId = "1";
+        addKnnDoc(INDEX_NAME, docOneId, FIELD_NAME, vectorForDocumentOne);
+
+        final Float[] vectorForDocumentTwo = {2.0f, 1.0f, 3.8f, 2.5f};
+        final String docTwoId = "2";
+        addKnnDoc(INDEX_NAME, docTwoId, FIELD_NAME, vectorForDocumentTwo);
+
+        //checking that one of two documents in retrievable based on closer similarity to query vector
+        int k = 1;
```

Comment:
I think we can just make k = 2 and then:
1. Make sure 2 docs are returned by the k-NN query
2. Update doc 2 to null
3. Assert the value returned from retrieving doc is null
4. Make sure 1 doc is returned by the k-NN query when k = 2 (doc id 1)
5. Update doc 2 back to the vector
6. Make sure retrieving doc 2 gives the correct vector
7. Make sure 2 docs are returned by the k-NN query

Reply:
Discussed offline, changing test to use k equals 2 instead of 1 for knn searches